### PR TITLE
Add DB_NAME to config file

### PR DESCRIPTION
**Pull request by jjeongatl**
## Pull request description

Previously we implemented the logic to dynamically decide db_name for RDS depending on whether db_snapshot_identifier was used or not. This is because of inconsistency between snapshot provided by dc-apt team and snapshot taken by the data-center-terraform. 
This change will allow users to use input whichever db_name matching with their RDS snapshot
## Checklist
- [ ] I have successful end to end tests run (with & without domain)
- [x] I have added unit tests (if applicable)
- [x] I have user documentation (if applicable)

### CODE REVIEW

File: config.tfvars [hunk 1]
```diff
@@ -76,6 +76,8 @@ jira_db_major_engine_version = "12"
 jira_db_instance_class       = "db.t3.micro"
 jira_db_allocated_storage    = 100
 jira_db_iops                 = 1000
+# IF you want to restore the database from a snapshot, please provide the default db name for the snapshot. If the snapshot does not have default db name comment out this variable.
```
Comment (nghazali) [inline config.tfvars:79]:
```suggestion
# If you restore the database, you need to provide the db name from the snapshot. 
```

File: config.tfvars [hunk 2]
```diff
@@ -132,6 +134,8 @@ confluence_db_major_engine_version = "11"
 confluence_db_instance_class       = "db.t3.micro"
 confluence_db_allocated_storage    = 100
 confluence_db_iops                 = 1000
+# IF you want to restore the database from a snapshot, please provide the default db name for the snapshot. If the snapshot does not have default db name comment out this variable.
```
Comment (nghazali) [inline config.tfvars:137]:
```suggestion
# If you restore the database, you need to provide the db name from the snapshot. 
```

File: config.tfvars [hunk 3]
```diff
@@ -206,6 +210,8 @@ bitbucket_db_major_engine_version = "13"
 bitbucket_db_instance_class       = "db.t3.micro"
 bitbucket_db_allocated_storage    = 100
 bitbucket_db_iops                 = 1000
+# IF you want to restore the database from a snapshot, please provide the default db name for the snapshot. If the snapshot does not have default db name comment out this variable.
```
Comment (nghazali) [inline config.tfvars:213]:
```suggestion
# If you restore the database, you need to provide the db name from the snapshot. 
```

File: modules/products/bamboo/main.tf [hunk 4]
```diff
@@ -24,4 +24,6 @@ module "database" {
   instance_class          = var.db_configuration["db_instance_class"]
   iops                    = var.db_configuration["db_iops"]
   vpc                     = var.vpc
+  db_name                 = var.db_configuration["db_name"]
+
```
Comment (nghazali) [inline modules/products/bamboo/main.tf:28]:
```suggestion
  db_name                 = var.db_configuration["db_name"]
```

Comment (nghazali) [reply modules/products/bamboo/main.tf:28]: removed an extra space line.

**Review by nghazali — APPROVED** [review]
Good improvement. Just left some suggestions.

File: variables.tf [hunk 5]
```diff
@@ -187,6 +187,12 @@ variable "jira_db_iops" {
   type        = number
 }
 
+variable "jira_db_name" {
+  description = "The default DB name of the DB instance."
+  default     = null
+  type        = string
+}
```
Comment (nghazali) [inline variables.tf:194]: we man can add validation for db name?

File: variables.tf [hunk 6]
```diff
@@ -693,6 +711,12 @@ variable "bamboo_db_iops" {
   type        = number
 }
 
+variable "bamboo_db_name" {
+  description = "The default DB name of the DB instance."
+  default     = null
+  type        = string
+}
+
```
Comment (nghazali) [inline variables.tf:719]: Do we need this for Bamboo? I think we use different approach rather than snapshot for Bamboo.

Comment (nghazali) [reply variables.tf:719]: But on the other hand, is good to be consistent.

File: variables.tf (hunk 5 above)
Comment (jjeongatl) [reply variables.tf:194]: added in the module level

File: variables.tf (hunk 6 above)
Comment (jjeongatl) [reply variables.tf:719]: This is optional but I added to keep it consistent

**Comment by nghazali** [comment]
@jjeongatl , before merge please run e2e test and complete the PR checklist

**Pull request merged**